### PR TITLE
iOS SDK Swift migration

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## Description of the change

> iOS SDK v2 (Swift migration) documentation.

## Type of documentation

- [x] New documentation
- [ ] Documentation update
- [ ] Hotfix

## Notion ticket link

> [iOS SDK Swift migration](https://www.notion.so/rudderstacks/Documentation-Update-Swift-SDK-c7e6e635bbdb4c91970926b20a3c8274)

### CODE REVIEW

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx [hunk 1]
```diff
@@ -0,0 +1,1114 @@
...
+
+The recommended and easiest way to add the iOS SDK to your project is through <code class="inline-code">Podfile</code>.
+</div>
+
+To install the iOS SDK depending on your preference, follow the steps below:
+
+<Tabs>
+  <TabList>
+    <Tab>CocoaPods</Tab>
+    <Tab>Carthage</Tab>
```
Comment (ameypv-rudder) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:48]: Will the Carthage installation steps remain the same?

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx [hunk 2]
```diff
@@ -0,0 +1,1114 @@
...
+    </TabPanels>
+</Tabs>
+
+The `screen` method accepts the following parameters:
+
+| Name         | Data type      | Presence | Description                                                              |
+| :----------- | :------------- | :------- | :--------------------------------------------- |
+| `screenName` | `NSString`     | Required      | Name of the screen viewed by the user.     |
+| `properties` | `NSDictionary` | Optional       | Extra property object to be passed along with the `screen` call. |
+| `options`    | `RudderOption` | Optional       | Extra options to be passed along with the `screen` call.     |
```
Comment (ameypv-rudder) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:416]: should this be `option` or `options`? As per the spec shared by you [here](https://www.notion.so/rudderstacks/Swift-migration-changes-Core-SDK-5ab9a96f526344d4bde3a149723cd18b) (refer to the Screen section)

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx [hunk 3]
```diff
@@ -0,0 +1,1114 @@
...
+
+| Name      | Data type      | Presence | Description                |
+| :-------- | :------------- | :------- | :------------------------------------ |
+| `groupId` | `String`       | Required      | The unique identifier of the group with which you want to associate your user.        |
+| `traits`  | `NSDictionary` | Optional       | Any other property of the organization you want to pass along with the call. |
+| `options` | `RudderOption` | Optional       | Extra event-level options to be passed along with the `group` call.              |
+
+<div class="warningBlock">
+
+The iOS SDK does not persist the group traits across the sessions.
```
Comment (ameypv-rudder) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:466]: What is the reason for this? I think this information will be beneficial.

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx [hunk 4]
```diff
@@ -0,0 +1,1114 @@
...
+client.reset()
+```
+</span>
+      </TabPanel>
+    </TabPanels>
+</Tabs>
+
+## Enabling/disabling user tracking via the `setOptOutStatus` API (GDPR support)
+
+RudderStack gives the users (for example, an EU user) the ability to opt out of tracking any user activity until they give their consent. You can do this by leveraging RudderStack's `setOptOutStatus` API.
```
Comment (ameypv-rudder) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:541]: Is the `setOptOutStatus` API available only from v2?

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx [hunk 5]
```diff
@@ -0,0 +1,1114 @@
...
+To migrate from an older version 1.0.2, update the usage of the following classes:
+
+| Previous Name | Updated Name |
+| :---------------| :---------------|
+| `RudderClient` | `RSClient`       |
+| `RudderConfig`| `RSConfig`     |
+| `RudderConfigBuilder` | `RSConfigBuilder` |
+| `RudderLogLevelDebug` | `RSLogLevelDebug` |
+
+### How do I ensure the events tracked just before closing/backgrounding the app are sent immediately and not on the next app launch?
```
Comment (ameypv-rudder) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:1042]: If `enableBackgroundMode` is deprecated in this version, should we remove this FAQ as well?

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx [hunk 6]
```diff
@@ -0,0 +1,1114 @@
...
+### Does the iOS SDK support the tvOS and watchOS platforms?
+
+Yes, the iOS SDK supports tvOS and watchOS platforms. Refer to the table below for the version details:
+
+| Platform | Minimum SDK version supported |
+| :--------| :----------------|
+| [tvOS](https://developer.apple.com/tvos/) | 1.1.0 |
+| [watchOS](https://developer.apple.com/watchos/) | 1.3.0 |
+
+### How do I migrate from v1.0.2?
```
Comment (ameypv-rudder) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:1031]: Please review this FAQ and confirm if the **updated** parameter names remain the same

**Comment by ameypv-rudder** [comment]
@pallabmaiti we would also need to review all the code snippets for accuracy and consistency with the v2 release.

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx [hunk 7]
```diff
@@ -0,0 +1,1114 @@
+---
+# slug: "/docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk"
+title: "iOS"
```
Comment (pallabmaiti) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:3]: Should be `Swift`.

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx [hunk 8]
```diff
@@ -0,0 +1,1114 @@
...
+description: >-
+  Detailed technical documentation on the RudderStack iOS SDK to send
```
Comment (pallabmaiti) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:5]:
```suggestion
  Detailed technical documentation on the RudderStack SDK to send
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx [hunk 9]
```diff
@@ -0,0 +1,1114 @@
...
+  events from your iOS apps to various destinations.
```
Comment (pallabmaiti) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:6]:
```suggestion
  events from your iOS/tvOS/watchOS/macOS apps to various destinations.
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx [hunk 10]
```diff
@@ -0,0 +1,1114 @@
...
+<div class="successBlock">
+
+You can now integrate the iOS SDK with your tvOS and watchOS apps and seamlessly track user events without the need for any additional configuration. The iOS SDK supports tvOS starting from version <strong>1.1.0</strong> and <strong>watchOS</strong> is supported starting from the SDK version <strong>1.3.1</strong>.
+</div>
+
+// Add stability beta badge here
+
+## SDK setup requirements
+
+To set up the RudderStack iOS SDK, the following prerequisites must be met:
```
Comment (pallabmaiti) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:24]:
```suggestion
To set up the RudderStack Swift SDK, the following prerequisites must be met:
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx [hunk 11]
```diff
@@ -0,0 +1,1114 @@
...
+
+- You will need to set up a [RudderStack account](https://app.rudderstack.com).
+- Set up an iOS source in the dashboard. For more information, follow the [Adding a Source and Destination in RudderStack](https://rudderstack.com/docs/connections/adding-source-and-destination-rudderstack/) guide.  You should then see a **Write Key** for this source, as shown:
+
+<img src="../../../assets/event-stream-sources/ios-sdk-1.png" alt="iOS source write key" />
+
+- You also need a data plane URL. Follow [**this section**](https://rudderstack.com/docs/rudderstack-open-source/installing-and-setting-up-rudderstack/#what-is-a-data-plane-url-where-do-i-get-it) for more information on the data plane URL and where to get it.
+- Finally, you will need a Mac with the latest version of [Xcode](https://developer.apple.com/xcode/).
+
+## Installing the RudderStack iOS SDK
```
Comment (pallabmaiti) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:34]:
```suggestion
## Installing the RudderStack Swift SDK
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx [hunk 12]
```diff
@@ -0,0 +1,1114 @@
...
+
+The RudderStack iOS SDK is distributed through [Cocoapods](https://cocoapods.org/pods/Rudder) and [Carthage](https://github.com/Carthage/Carthage).
```
Comment (pallabmaiti) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:36]:
```suggestion
The RudderStack SDK is distributed through [Cocoapods](https://cocoapods.org/pods/Rudder) and [Carthage](https://github.com/Carthage/Carthage).
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx [hunk 13]
```diff
@@ -0,0 +1,1114 @@
...
+
+<div class="successBlock">
+
+The recommended and easiest way to add the iOS SDK to your project is through <code class="inline-code">Podfile</code>.
+</div>
+
+To install the iOS SDK depending on your preference, follow the steps below:
```
Comment (pallabmaiti) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:43]:
```suggestion
To install the SDK depending on your preference, follow the steps below:
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx [hunk 14]
```diff
@@ -0,0 +1,1114 @@
...
+
+<Tabs>
+  <TabList>
+    <Tab>CocoaPods</Tab>
+    <Tab>Carthage</Tab>
+  </TabList>
+  <TabPanels>
+    <TabPanel>
+      <ul>
+      <li>Add the iOS SDK to your <code class="inline-code">Podfile</code>, as shown:
```
Comment (pallabmaiti) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:53]:
```suggestion
      <li>Add the SDK to your <code class="inline-code">Podfile</code>, as shown:
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx [hunk 15]
```diff
@@ -0,0 +1,1114 @@
...
+<span>
+
+```ruby
+pod 'RudderStack', '~> 1.0.0-beta.3'
```
Comment (pallabmaiti) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:57]:
```suggestion
pod 'Rudder', '~> 2.0.0'
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx [hunk 16]
```diff
@@ -0,0 +1,1114 @@
...
+  <TabList>
+    <Tab>Objective-C</Tab>
+    <Tab>Swift</Tab>
+  </TabList>
+    <TabPanels>
+      <TabPanel>
+<span>
+
+```objectivec
+#import <Rudder/Rudder.h>
```
Comment (pallabmaiti) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:106]:
```suggestion
@import Rudder;
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx [hunk 17]
```diff
@@ -0,0 +1,1114 @@
...
+      <TabPanel>
+<span>
+
+```objectivec
+RSConfig *config = [[RSConfig alloc] initWithWriteKey:WRITE_KEY];
+[config dataPlaneURL:DATA_PLANE_URL];
+[config trackLifecycleEvents:YES];
+[config recordScreenViews:YES];
+
+[[RSClient sharedInstance] configureWith:config]
```
Comment (pallabmaiti) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:145]:
```suggestion
[[RSClient sharedInstance] configureWith:config];
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx [hunk 18]
```diff
@@ -0,0 +1,1114 @@
...
+You can disable these events using the <code class="inline-code">trackLifecycleEvents</code> method of <code class="inline-code">RSConfig</code> and passing <code class="inline-code">false</code>. However, it is highly recommended to keep them enabled.
+</div>
+
+## Configuring the RudderStack client
+
+You can configure your client based on the following parameters using `RudderConfig`:
+
+| Parameter               | Type      | Description    | Default value      |
+| :---------------------- | :-------- | :-------------- | :------- |
+| `logLevel`     | `RSLogLevel`     | Controls how much of the log you want to see from the iOS SDK.                | `RSLogLevel.None`                                                           |
```
Comment (pallabmaiti) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:188]:
```suggestion
| `logLevel`     | `RSLogLevel`     | Controls how much of the log you want to see from the SDK.                | `RSLogLevel.none`                                                           |
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx [hunk 19]
```diff
@@ -0,0 +1,1114 @@
...
+| `recordScreenViews`     | Boolean | Determines if the SDK will capture will capture screen view events automatically.   | `false`   |
+| `controlPlaneUrl`       | String  | This parameter should be changed **only if** you are self-hosting the control plane. Check the [Self-hosted control plane](https://rudderstack.com/docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/#self-hosted-control-plane) section below for more information. The SDK will add `/sourceConfig` along with this URL to fetch the required configuration. | `https://api.rudderlabs.com`   |
+
+### Self-hosted control plane
+
+If you are using a device mode destination like Adjust, Firebase, etc., the iOS SDK needs to fetch the required configuration from the control plane. If you are using the [Control Plane Lite](https://www.rudderstack.com/docs/rudderstack-open-source/control-plane-lite/) utility to host your own control plane, then follow [this section](https://www.rudderstack.com/docs/rudderstack-open-source/control-plane-lite/#control-plane-url) to specify `controlPlaneUrl` in `RudderConfig` that points to your hosted source configuration file.
```
Comment (pallabmaiti) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:200]:
```suggestion
If you are using a device mode destination like Adjust, Firebase, etc., the SDK needs to fetch the required configuration from the control plane. If you are using the [Control Plane Lite](https://www.rudderstack.com/docs/rudderstack-open-source/control-plane-lite/) utility to host your own control plane, then follow [this section](https://www.rudderstack.com/docs/rudderstack-open-source/control-plane-lite/#control-plane-url) to specify `controlPlaneUrl` in `RSConfig` that points to your hosted source configuration file.
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx [hunk 20]
```diff
@@ -0,0 +1,1114 @@
...
+<li><code class="inline-code">Application Updated</code></li>
+<li><code class="inline-code">Application Opened</code></li>
+<li><code class="inline-code">Application Backgrounded</code></li>
+</ul>
+You can disable these events using the <code class="inline-code">trackLifecycleEvents</code> method of <code class="inline-code">RSConfig</code> and passing <code class="inline-code">false</code>. However, it is highly recommended to keep them enabled.
+</div>
+
+## Configuring the RudderStack client
+
+You can configure your client based on the following parameters using `RudderConfig`:
```
Comment (pallabmaiti) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:184]:
```suggestion
You can configure your client based on the following parameters using `RSConfig`:
```

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx [hunk 21]
```diff
@@ -0,0 +1,1114 @@
...
+    </TabPanels>
+</Tabs>
+
+The `identify` method accepts the following parameters:
+
+| Name      | Data type      | Presence | Description         |
+| :-------- | :------------- | :------- | :--------------------------- |
+| `userId`  | `NSString`     | Required      | Uniquely identifies the visiting user.   |
+| `traits`  | `NSDictionary` | Optional       | Information on the user traits. Use the `dict` method of `RudderTraits` to convert to `NSDictionary` easily. |
+| `options` | `RudderOption` | Optional       | Extra options for the `identify` event.      |
```
Comment (pallabmaiti) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:258]:
```suggestion
| `option` | `RSOption` | Optional       | Extra options for the `identify` event.      |
```

**Comment by pallabmaiti** [comment]
All the `RudderOption` will be `RSOption`, `options` to `option`, `RudderConfig` to `RSConfig`

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx [hunk 22]
```diff
@@ -0,0 +1,1114 @@
...
+    </TabPanels>
+</Tabs>
+
+## FAQs
+
+### I'm facing issues building with Carthage on XCode 12. What should I do?
+
+If you're facing an issue with Carthage and XCode 12, you can follow [this workaround](https://github.com/Carthage/Carthage/blob/master/Documentation/Xcode12Workaround.md) suggested by the Carthage team.
+
+### Does the iOS SDK support the tvOS and watchOS platforms?
```
Comment (pallabmaiti) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:1022]: Add macOS here.

File: stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx [hunk 23]
```diff
@@ -0,0 +1,1114 @@
...
+
+### How do I migrate from v1.0.2?
+
+To migrate from an older version 1.0.2, update the usage of the following classes:
+
+| Previous Name | Updated Name |
+| :---------------| :---------------|
+| `RudderClient` | `RSClient`       |
+| `RudderConfig`| `RSConfig`     |
+| `RudderConfigBuilder` | `RSConfigBuilder` |
```
Comment (pallabmaiti) [inline stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:1039]: `RSConfigBuilder` - this class is not available.

**Review by pallabmaiti — CHANGES_REQUESTED** [review]

**Comment by ameypv-rudder** [comment]
Because of some issues with rebasing this branch with `main`, I've created a fresh PR with all the changes incorporated from @pallabmaiti 's feedback. Hence, closing this PR.

**Pull request closed without merging**